### PR TITLE
Fixes issue with `showAttachments` on dynamic layers

### PR DESCRIPTION
**Pull request by tmcgee**
Previously, this code was not executing when the layerIds array did not exist in the layer's configuration.

Also added a test for the `showAttachments` config property so that feature layers are added only for those layers with that property.

### CODE REVIEW

File: viewer/js/gis/dijit/Identify.js [hunk 1]
```diff
@@ -154,8 +154,11 @@ define([
                 } else if (layer.layerInfos) {
                     array.forEach(layer.layerInfos, lang.hitch(this, function (subLayerInfo) {
                         var subLayerId = subLayerInfo.id;
-                        if ((layerInfo.layerIds === null) || (array.indexOf(layerInfo.layerIds, subLayerId) >= 0)) {
-                            this.getFeatureLayerForDynamicSublayer(layer, subLayerId);
+                        if ((typeof layerInfo.layerIds === 'undefined') || (array.indexOf(layerInfo.layerIds, subLayerId) >= 0)) {
+                            infoTemplate = this.getInfoTemplate(layer, subLayerId);
```
Comment (tmcgee) [inline viewer/js/gis/dijit/Identify.js]: This additional check may be problematic. I am looking into it.

Comment (tmcgee) [reply viewer/js/gis/dijit/Identify.js]: I removed this check for now as it causes some misbehavior. Removing it doesn't affect the intended bug fix.

**Review by amcgee — APPROVED** [review]
Assuming `array.indexOf` doesn't break when passed a `null` array this should be fine.

**Comment by tmcgee** [comment]
Dojo's `array` handles that scenario.

**Pull request merged**